public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx

```diff
@@ -0,0 +1,650 @@
+    constructor(props: Props) {
+        super(props);
+        this.changeResults = this.changeResults.bind(this);
+        // TODO add url params and change them based on search value
+        this.state = {
+            domain: null,
+            totalParticipants: 0,
+            title: '',
+            subTitle: '',
+            searchWord: this.props.searchTerm,
+            showStatement: false,
+            top10Results: null,
+            selectedConcept: null,
+            numPages: null,
+            totalResults: null,
+            concepts: [],
+            standardConcepts: [],
+            currentPage: 1,
+            loading: true,
+            medlinePlusLink: null,
+            medlineTerm: null,
+            selectedMeasurementTypeFilter: false,
+            measurementTestFilter: true,
+            measurementOrderFilter: true,
+            showTopConcepts: true,
+            fromDifferentPage: false
+        };
+   }
+
+   domainTotals = _.debounce(() => {
+             this.getDomainTotals();
+             this.getTopConcepts();
+             this.changeUrl();
+           }, 1000);
+
+   componentDidMount() {
+        this.getDomainTotals();
+   }
+
+   componentDidUpdate(prevProps: Readonly<Props>) {
+        if (prevProps.domainId !== this.props.domainId) {
+            this.getDomainTotals();
+        }
+   }
+
+   changeUrl() {
+    let url = 'ehr/' + this.props.domainId;
+    if (this.state.searchWord) {
+        url += '?search=' + this.state.searchWord;
+    }
+    window.history.replaceState(null, 'Ehr View', url);
+   }
+
+   getDomainTotals() {
+    const {searchWord, measurementTestFilter, measurementOrderFilter} = this.state;
+    dataBrowserApi().getDomainTotals(searchWord, measurementTestFilter ? 1 : 0, measurementOrderFilter ? 1 : 0)
+                .then(results => {
+                    results.domainInfos.forEach(domain => {
+                        const thisDomain = Domain[domain.domain];
+                        if (thisDomain && thisDomain.toLowerCase() === this.props.domainId) {
+                            const ehrDomain = domain;
+                            const subTitle = 'Keyword: ' + this.props.searchTerm;
+                            const title = ehrDomain.name;
+                            const totalParticipants = ehrDomain.participantCount;
+                            const numPages = Math.ceil(ehrDomain.standardConceptCount / 50);
+                            this.setState({
+                                domain: domain,
+                                title: title,
+                                subTitle: subTitle,
+                                totalParticipants: totalParticipants,
+                                numPages: numPages,
+                                totalResults: ehrDomain.standardConceptCount,
+                            }, () => {
+                                this.getTopConcepts();
+                            });
```

Comment:
Nit: these vars only get used once, can be simplified if we drop them:
```suggestion
                            this.setState({
                                domain: domain,
                                title: domain.name,
                                subTitle: 'Keyword: ' + this.props.searchTerm,
                                totalParticipants: domain.participantCount,
                                numPages: Math.ceil(domain.standardConceptCount / 50),
                                totalResults: domain.standardConceptCount,
                            }, () => {
                                this.getTopConcepts();
                            });
```

Reply:
Changed this thanks